src/equations/covariant_advection.jl

```diff
@@ -0,0 +1,99 @@
+@muladd begin
+#! format: noindent
+
+@doc raw"""
+    CovariantLinearAdvectionEquation2D <: AbstractCovariantEquations{2, 3, 3}
```

Comment:
Could you please add a reference (book, article, or report) that provides more information on equations in covariant form? This would really help me review the new routines and be a great resource for future TrixiAtmo users too.

Reply:
Done in [d515349](https://github.com/trixi-framework/TrixiAtmo.jl/pull/31/commits/d515349a2c81deaa3cd1212ecfc771d60fdc2dd8).

Reply:
Thanks!